### PR TITLE
[#34] Better Notification System

### LINKED ISSUE #34: Better Notification System
The current implementation of the notifications is difficult to use. For this reason, a newer and better documented API should be developed.

### PR DESCRIPTION
**Pull request by TheMeinerLP**
## Proposed changes

Fixes #34

This PR aims to create a new interface to send better notifications. Old code was removed from the demo and an obsolescence message was added to the old notification system. 
The new design is based on a Creation Pattern or Buidler Pattern

## Types of changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of
them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before
merging your code._

- [X] I have read the CONTRIBUTING.md
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you
did and what alternatives you considered, etc...

### CODE REVIEW

File: src/main/java/net/minestom/server/notifications/Notification.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        Builder icon(@NotNull ItemStack itemStack);
+
+        /**
+         * Returns an instance of the creation notification
+         * @return the instance
+         */
+        Notification build();
+    }
+
+
```
Comment (theEvilReaper) [inline src/main/java/net/minestom/server/notifications/Notification.java]: Unnecessary empty lines

File: src/main/java/net/minestom/server/notifications/NotificationBuilder.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+/**
+ * {@inheritDoc}
+ */
+final class NotificationBuilder implements Notification.Builder {
+    private Component title;
+    private FrameType type;
+    private ItemStack icon;
+
+
```
Comment (theEvilReaper) [inline src/main/java/net/minestom/server/notifications/NotificationBuilder.java]: Unnecessary empty line

File: src/main/java/net/minestom/server/notifications/Notification.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    /**
+     * Creates a new builder instance
+     * @return an instance of the builder
+     */
+    @Contract(pure = true)
+    static @NotNull Builder builder() {
+        return new NotificationBuilder();
+    }
+
```
Comment (theEvilReaper) [inline src/main/java/net/minestom/server/notifications/Notification.java:44]: Unnecessary empty line

File: src/main/java/net/minestom/server/notifications/Notification.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+     */
+    void send(@NotNull Player player);
+
+    /**
+     * Send the notification to a collection of clients
+     * @param players to get be sent
+     */
+    void send(@NotNull Collection<@NotNull Player> players);
+
+
```
Comment (theEvilReaper) [inline src/main/java/net/minestom/server/notifications/Notification.java]: Unnecessary empty line

**Review by theEvilReaper — CHANGES_REQUESTED** [review]

**Review by Codestech1 — APPROVED** [review]

**Review by theEvilReaper — APPROVED** [review]

**Pull request merged**